### PR TITLE
services/horizon/ingest: removed legacy core cursor update against during ledger ingestion

### LINKED ISSUE #5153: No need to update cursor when ingesting from captive core
As @sreuland reported, we always update the cursor via the stellar core client in the resume state:

https://github.com/stellar/go/blob/master/services/horizon/internal/ingest/fsm.go#L526

We should only be doing this if we're running legacy ingestion via the stellar core postgres DB. When running captive core, there is no need to update the cursor.

### PR DESCRIPTION
**Pull request by sreuland**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Removed the core cursor update that is attempted by ingest fsm on each ledger, as it was only relevant to legacy core db ingestion which is no longer enabled, only captive core ingestion.

Removed the configuration flags `--stellar-core-url-db`, `--cursor-name` `--skip-cursor-update` , as they are related to legacy core db ingestion which is no longer usable. 

if one of the aforementioned flags is present, horizon will exit and report the error on console
```
$ DATABASE_URL=dburl STELLAR_CORE_DATABASE_URL=xxx ./horizon
flag --stellar-core-db-url and environment variable STELLAR_CORE_DATABASE_URL have been removed and no longer valid, must use captive core configuration for ingestion
```

Closes #5153 

### Why

discovered the un-necessary core cursor update taking time during performance benchmarking.

### CODE REVIEW

File: services/horizon/internal/ingest/main.go [hunk 1]
```diff
@@ -273,13 +272,6 @@ func NewSystem(config Config) (System, error) {
 			cancel()
 			return nil, errors.Wrap(err, "error creating captive core backend")
 		}
-	} else {
```
Comment (sreuland) [inline services/horizon/internal/ingest/main.go:276]: this was unreachable code, `config.RemoteCaptiveCoreEnabled()` is the inverse of `config.LocalCaptiveCoreEnabled()`, horizon assumes only captive core.

File: services/horizon/internal/ingest/main.go [hunk 2]
```diff
@@ -252,7 +250,8 @@ func NewSystem(config Config) (System, error) {
 			cancel()
 			return nil, errors.Wrap(err, "error creating captive core backend")
 		}
-	} else if config.LocalCaptiveCoreEnabled() {
+	} else {
```
Comment (urvisavla) [inline services/horizon/internal/ingest/main.go:253]: I think we can remove lines 247 to 252 as we've deprecated remote captive core as well.

**Review by urvisavla — APPROVED** [review]
lgtm!

File: services/horizon/internal/ingest/main.go (hunk 2 above)
Comment (sreuland) [reply services/horizon/internal/ingest/main.go]: sure, I swept out the remnants of that - https://github.com/stellar/go/pull/5158/commits/a781a11ea006e418491043bb5482fed073ebcb34

Comment (urvisavla) [reply services/horizon/internal/ingest/main.go]: Didn't realize it was still left in multiple places. Thanks for cleaning it up!

**Pull request merged**